src/Hid.Shared/Hid.Helpers.cs

```
@@ -0,0 +1,137 @@
+    {
+        public static Guid HidD_GetHidGuid()
+        {
+            Guid guid;
+            HidD_GetHidGuid(out guid);
+            return guid;
+        }
+
+        public static HiddAttributes HidD_GetAttributes(SafeFileHandle hFile)
+        {
+            var result = HiddAttributes.Create();
+            if (!HidD_GetAttributes(hFile, ref result))
+            {
+                throw new Win32Exception();
+            }
+
+            return result;
+        }
+
+        public static SafePreparsedDataHandle HidD_GetPreparsedData(SafeFileHandle hDevice)
+        {
+            SafePreparsedDataHandle preparsedDataHandle;
+            if (!HidD_GetPreparsedData(hDevice, out preparsedDataHandle))
+            {
+                throw new Win32Exception();
+            }
+
+            return preparsedDataHandle;
+        }
+
+        public static HidpCaps HidP_GetCaps(SafePreparsedDataHandle preparsedData)
+        {
+            var hidCaps = default(HidpCaps);
+            var result = HidP_GetCaps(preparsedData, ref hidCaps);
+            switch (result)
+            {
+                case NTStatus.HIDP_STATUS_SUCCESS:
+                    return hidCaps;
+
+                case NTStatus.HIDP_STATUS_INVALID_PREPARSED_DATA:
+                    throw new ArgumentException("The specified preparsed data is invalid.", nameof(preparsedData));
+
+                default:
+                    result.ThrowOnError();
+                    return default(HidpCaps);
+            }
+        }
+
+        public static bool HidD_GetManufacturerString(SafeFileHandle hidDeviceObject, out string result)
+        {
+            return GrowBuffer(sb => HidD_GetManufacturerString(hidDeviceObject, sb, sb.Capacity), out result);
+        }
+
+        public static bool HidD_GetProductString(SafeFileHandle hidDeviceObject, out string result)
+        {
+            return GrowBuffer(sb => HidD_GetProductString(hidDeviceObject, sb, sb.Capacity), out result);
```

Do you want to make these helper methods return `string` and throw instead of return bool and use an out parameter?